src/units/abilities.cpp

```diff
@@ -1591,6 +1587,17 @@ bool attack_type::special_active_impl(const_attack_ptr self_attack, const_attack
 	const_attack_ptr att_weapon = is_attacker ? self_attack : other_attack;
 	const_attack_ptr def_weapon = is_attacker ? other_attack : self_attack;
 
+	// Filter firststrike here, if both units have first strike then the effects cancel out. Only check
+	// the opponent if "whom" is the defender, otherwise this leads to infinite
+	// recursion.
```

Comment:
Put "recursion" on the line above - the first comment line is already longer anyway.

Reply:
Done.